pageserver/src/http/routes.rs

```diff
@@ -3742,7 +3765,11 @@ pub fn make_router(
         )
         .put(
             "/v1/tenant/:tenant_shard_id/timeline/:timeline_id/detach_ancestor",
-            |r| api_handler(r, timeline_detach_ancestor_handler),
+            |r| api_handler(r, timeline_detach_ancestor_handler_v1),
+        )
+        .put(
+            "/v1/tenant/:tenant_shard_id/timeline/:timeline_id/detach_ancestor_v2",
```

Comment:
I think people have just used "v2" as a shorthand when planning, let's give this a semantically meaningful name, or even just let people specify a parameter to detach_ancestor that controls whether to reparent things.

Reply:
Yes I agree, my personal preference would be having a detach behavior in the API instead of having a new one. I can remove this newly-added API and add a query parameter to specify the detach behavior.